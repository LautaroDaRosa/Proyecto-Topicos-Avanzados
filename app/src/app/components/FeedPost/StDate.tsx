import styled from 'styled-components';

const StDate = styled.span`
  color: #7e7e7e;
  font-size: 15px;
  font-style: normal;
  font-weight: 300;
  line-height: normal;
  text-transform: capitalize;
`;

export default StDate;
