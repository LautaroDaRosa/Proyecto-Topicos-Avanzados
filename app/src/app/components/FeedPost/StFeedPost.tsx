import styled from 'styled-components';

const StFeedPost = styled.div`
  display: flex;
  border-radius: 8px;
  background: white;
  box-shadow: 0px 4px 9px 0px rgba(0, 0, 0, 0.1);
  width: 100%;
  padding: 20px;
  flex-direction: column;
  align-items: flex-start;
  gap: 20px;
  font-size: 18px;
  line-height: normal;
`;

export default StFeedPost;
