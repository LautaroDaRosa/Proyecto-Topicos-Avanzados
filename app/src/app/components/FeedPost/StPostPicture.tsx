import styled from 'styled-components';

const StPostPicture = styled.img`
  width: 100%;
`;

export default StPostPicture;
