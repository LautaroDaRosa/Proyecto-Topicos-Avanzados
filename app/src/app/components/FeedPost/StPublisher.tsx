import styled from 'styled-components';

const StPusblisher = styled.div`
  display: flex;
  gap: 10px;

  div {
    display: flex;
    flex-direction: column;
  }

  a {
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }
`;

export default StPusblisher;
