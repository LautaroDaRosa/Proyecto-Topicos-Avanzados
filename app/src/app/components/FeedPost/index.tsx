import StFeedPost from './StFeedPost';
import StPusblisher from './StPublisher';
import { Link } from 'react-router-dom';
import ProfileImage from '../ProfileImage';
import StPostPicture from './StPostPicture';
import StDate from './StDate';
import { Post } from 'types';

const getFormattedDate = (date: string) => {
  return date
    ? new Intl.DateTimeFormat('en-US', {
        dateStyle: 'long',
      }).format(new Date(date))
    : '-';
};

const FeedPost = ({ id, content, photo, createdAt, user }: Post) => {
  return (
    <StFeedPost>
      <StPusblisher>
        <ProfileImage type="post-publisher" src={user.photo} alt="User pic" />
        <div>
          <Link to={`/profile/${user.id}`} id={`user-${user.id}`}>
            <span>{user.fullname}</span>
          </Link>
          <StDate>{getFormattedDate(createdAt)}</StDate>
        </div>
      </StPusblisher>
      <span>{content}</span>
      {photo && <StPostPicture src={photo} alt="Post pic!" />}
    </StFeedPost>
  );
};

export default FeedPost;
