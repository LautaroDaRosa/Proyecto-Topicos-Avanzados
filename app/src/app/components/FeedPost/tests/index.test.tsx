import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import FeedPost from '../index';

describe('Feed Post', () => {
  it('renders post data correctly', () => {
    const postInfo = {
      id: 1,
      content: 'Quien esta para Lokotas?',
      photo:
        'https://lokotas.com.uy/wp-content/uploads/2019/04/iso-lokotas.png',
      userFullname: 'Ramiro Nieto',
      userPhoto: 'employees/rnieto.png',
      datetime: '2020-07-21 19:18:03.496513+00',
    };

    const { container } = render(
      <Router>
        <FeedPost
          id={postInfo.id}
          content={postInfo.content}
          photo={postInfo.photo}
          key={postInfo.id}
          user={{
            id: 1,
            fullname: postInfo.userFullname,
            photo: postInfo.userPhoto,
          }}
          createdAt={postInfo.datetime}
        />
      </Router>,
    );

    const postPic = screen.getByRole('img', { name: 'Post pic!' });
    const userPic = screen.getByRole('img', { name: 'User pic' });
    screen.getByText('Ramiro Nieto');
    screen.getByText('July 21, 2020');
    screen.getByText('Quien esta para Lokotas?');

    expect(userPic).toHaveAttribute('src', 'employees/rnieto.png');
    expect(postPic).toHaveAttribute(
      'src',
      'https://lokotas.com.uy/wp-content/uploads/2019/04/iso-lokotas.png',
    );

    expect(container).toMatchSnapshot();
  });
});
