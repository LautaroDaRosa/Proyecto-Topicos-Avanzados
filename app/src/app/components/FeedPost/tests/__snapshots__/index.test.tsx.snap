// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed Post renders post data correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 8px;
  background: white;
  box-shadow: 0px 4px 9px 0px rgba(0,0,0,0.1);
  width: 100%;
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 20px;
  font-size: 18px;
  line-height: normal;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
}

.c1 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  object-fit: cover;
  height: 48px;
  width: 48px;
  border-radius: 50%;
}

.c4 {
  width: 100%;
}

.c3 {
  color: #7e7e7e;
  font-size: 15px;
  font-style: normal;
  font-weight: 300;
  line-height: normal;
  text-transform: capitalize;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="User pic"
        class="c2"
        src="employees/rnieto.png"
        type="post-publisher"
      />
      <div>
        <a
          href="/profile/1"
          id="user-1"
        >
          <span>
            Ramiro Nieto
          </span>
        </a>
        <span
          class="c3"
        >
          July 21, 2020
        </span>
      </div>
    </div>
    <span>
      Quien esta para Lokotas?
    </span>
    <img
      alt="Post pic!"
      class="c4"
      src="https://lokotas.com.uy/wp-content/uploads/2019/04/iso-lokotas.png"
    />
  </div>
</div>
`;
